scripts/gas-refund-program/transactions-indexing/successful-swap-tx-indexing.ts

```diff
@@ -87,72 +88,92 @@ export async function computeSuccessfulSwapsTxFeesRefund({
       `fetched ${swaps.length} swaps withing startBlock ${_startBlock} and endBlock ${_endBlock}`,
     );
 
-    accumulatedTxFeesByAddress = swaps.reduce<TxFeesByAddress>((acc, swap) => {
-      const address = swap.txOrigin;
+    const swapsWithGasUsed = await Promise.all(
+      swaps.map(async swap => ({
+        ...swap,
+        txGasUsed: await getTransactionGasUsed({
+          chainId,
+          txHash: swap.txHash,
+        }),
+      })),
+    );
+
+    accumulatedTxFeesByAddress = swapsWithGasUsed.reduce<TxFeesByAddress>(
+      (acc, swap) => {
+        const address = swap.txOrigin;
+
+        const swapperAcc = acc[address];
+
+        const pspRateSameDay = findSameDayPrice(swap.timestamp);
+
+        if (!pspRateSameDay) {
+          logger.warn(
+            `Fail to find price for same day ${
+              swap.timestamp
+            } and rates=${JSON.stringify(
+              pspNativeCurrencyDailyRate.flatMap(p => p.timestamp),
+            )}`,
+          );
+
+          return acc;
+        }
+
+        const currGasUsed = new BigNumber(swap.txGasUsed);
+        const accumulatedGasUsed = currGasUsed.plus(
+          swapperAcc?.accumulatedGasUsed || 0,
+        );
 
-      const swapperAcc = acc[address];
+        const currGasUsedChainCur = currGasUsed.multipliedBy(
+          swap.txGasPrice.toString(),
+        ); // in wei
 
-      const pspRateSameDay = findSameDayPrice(swap.timestamp);
+        const accumulatedGasUsedChainCurrency = currGasUsedChainCur.plus(
+          swapperAcc?.accumulatedGasUsedChainCurrency || 0,
+        );
+
+        const currGasFeePSP = currGasUsedChainCur.dividedBy(pspRateSameDay);
+
+        const accumulatedGasUsedPSP = currGasFeePSP.plus(
+          swapperAcc?.accumulatedGasUsedPSP || 0,
+        );
 
-      if (!pspRateSameDay) {
-        logger.warn(
-          `Fail to find price for same day ${
-            swap.timestamp
-          } and rates=${JSON.stringify(
-            pspNativeCurrencyDailyRate.flatMap(p => p.timestamp),
-          )}`,
+        const totalStakeAmountPSP = stakes[address];
```

Comment:
I'm not sure it's worth to do now, but because `tsconfig.json` in the repo root is not very strict, TS here doesn't warn you about `totalStakeAmountPSP` being possibly `undefined`, and lots of other places too, I imagine.

If you want to at least enable that kind of check for `scripts` folder add `scripts/tsconfig.josn` with
```json
{
  "extends": "../tsconfig.json",
  "compilerOptions": {
    "strict": true,
    "noUncheckedIndexedAccess": true
  },
  "include": ["."]
}
```

But it may be a hassle and a waste of time to fix all TS warning that will appear. But more type safety. Up to you.

Reply:
I actually went there too by myself, then I had couple of stuffs to address, I thought it wasn't reasonable so I stopped ^^